core/src/main/scala/spire/optional/mapIntIntPermutation.scala

```diff
@@ -1,52 +0,0 @@
-package spire
-package optional
-
-import scala.collection.compat._
-import spire.scalacompat.SeqLike
-
-import spire.algebra.{Action, Group}
-import spire.algebra.partial.PartialAction
-import spire.syntax.cfor._
-import spire.syntax.group._
-import spire.util._
-
-final class MapIntIntIntAction extends Action[Int, Map[Int, Int]] {
-  def actr(k: Int, perm: Map[Int, Int]): Int = perm.getOrElse(k, k)
-  def actl(perm: Map[Int, Int], k: Int): Int = perm.find(_._2 == k).fold(k)(_._1)
-}
-
-final class MapIntIntGroup extends Group[Map[Int, Int]] {
-  def empty: Map[Int, Int] = Map.empty[Int, Int]
-  def combine(x: Map[Int, Int], y: Map[Int, Int]): Map[Int, Int] = {
-    val preimages = x.keys ++ y.keys
-
-    preimages.foldLeft(Map.empty[Int, Int]) {
-      case (prevMap, preimage) =>
-        val imageX = x.getOrElse(preimage, preimage)
-        val image = y.getOrElse(imageX, imageX)
-        if (preimage != image) prevMap + ((preimage, image)) else prevMap
-    }
-  }
-  def inverse(a: Map[Int, Int]): Map[Int, Int] = a.map(_.swap).toMap
-}
-
-final class MapIntIntSeqPartialAction[A, SA <: SeqLike[A, SA]](implicit cbf: Factory[A, SA]) extends PartialAction[SA, Map[Int, Int]] {
-  import mapIntIntPermutation._
-  def partialActl(perm: Map[Int, Int], sa: SA): Opt[SA] = {
-    if (perm.isEmpty) return Opt(sa)
-    if (perm.keys.max >= sa.size) return Opt.empty[SA]
```

Comment:
This had a bug, since negative keys would fail at runtime with `IndexOutOfBound`.

This also means we should decide whether negatives are allowed in Perms. The current Gen/Arb implementation doesn't generate them. I'm inclined to disallow them; any permutation ought to be able to permute a seq.

Reply:
restricting it to integers >= 0 also seems reasonable to me, fwiw

Reply:
That's consistent with other computational group theory systems, so I'll make it a constraint in the constructor :+1: